src/util.rs

```diff
@@ -22,23 +22,47 @@ pub const BITS_PER_BYTE: u32 = 8;
 /// that are not valid in Rust ident
 const BLACKLIST_CHARS: &[char] = &['(', ')', '[', ']', '/', ' ', '-'];
 
+#[cfg_attr(feature = "bin", derive(serde::Deserialize))]
```

Comment:
can we cfg this on `serde` instead

Reply:
I've tried to use `serde`, but it doesn't compile. I don't know why.

Reply:
it's probably because it's used as `dep:serde`, instead only do `serde` in the `features.bin = [...]`